### PR TITLE
ventoy-bin: 1.0.56 -> 1.0.72

### PR DESCRIPTION
**Pull request by k4leg**
###### Motivation for this change

Ref: https://github.com/NixOS/nixpkgs/issues/158264

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 158365` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ventoy-bin</li>
  </ul>
</details>

**Review by Tungsten842 — APPROVED** [review]
Works as intended, windows and Fedora ISOs boot without issues.

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 158365` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ventoy-bin</li>
  </ul>
</details>

File: pkgs/tools/cd-dvd/ventoy-bin/default.nix [hunk 1]
```diff
@@ -1,48 +1,70 @@
...
-}.${stdenv.hostPlatform.system} or (throw "Unsupported platform ${stdenv.hostPlatform.system}");
-defaultGuiType = if withGtk3 then "gtk3"
-                 else if withQt5 then "qt5"
-                 else "";
+assert withCryptsetup -> cryptsetup != null;
+assert withXfs -> xfsprogs != null;
+assert withExt4 -> e2fsprogs != null;
+assert withNtfs -> ntfs3g != null;
+assert withGtk3 -> gtk3 != null;
+assert withQt5 -> qt5 != null;
```
Comment (SuperSandro2000) [inline pkgs/tools/cd-dvd/ventoy-bin/default.nix:19]:
```suggestion
```

File: pkgs/tools/cd-dvd/ventoy-bin/default.nix [hunk 2]
```diff
@@ -1,48 +1,70 @@
...
-, withGtk3 ? true, gtk3
-, withQt5 ? false, qt5
+, bash, coreutils, dosfstools, exfat, gawk, gnugrep, gnused, hexdump, parted
+, procps, util-linux, which, xz
+, withCryptsetup ? false, cryptsetup ? null
+, withXfs ? false, xfsprogs ? null
+, withExt4 ? false, e2fsprogs ? null
+, withNtfs ? false, ntfs3g ? null
+, withGtk3 ? false, gtk3 ? null
+, withQt5 ? false, qt5 ? null
```
Comment (SuperSandro2000) [inline pkgs/tools/cd-dvd/ventoy-bin/default.nix:10]:
```suggestion
, withCryptsetup ? false, cryptsetup
, withXfs ? false, xfsprogs
, withExt4 ? false, e2fsprogs
, withNtfs ? false, ntfs3g
, withGtk3 ? false, gtk3
, withQt5 ? false, qt5
```
not necessary

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 158365` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ventoy-bin</li>
  </ul>
</details>

**Pull request merged**